### PR TITLE
[RUB-352] Rust: harden block tx_count parser admission tests + fuzz seeds

### PR DESCRIPTION
**Pull request by 2tbmz9y2xt-lang**
## Summary
- Add 4 regression tests covering hostile tx_count parser inputs (u64_max EOF, 2^32 EOF, trailing bytes, malformed CompactSize).
- Add 3 fuzz tests for tx_count boundary values: 0, 1, 252, 253, 65535, 2^32, u64_max.
- No production code touched; parser behavior unchanged.
- Non-consensus hardening only.

### CODE REVIEW

File: clients/rust/crates/rubin-consensus/tests/block_basic_fuzz.rs [hunk 1]
```diff
@@ -131,6 +131,45 @@ fn block_parse_deterministic() {
...
+    // 253: 0xFD tag + u16 LE = 0xFD, 0xFD, 0x00
+    // 65535: 0xFD tag + u16 LE = 0xFD, 0xFF, 0xFF
+    let seeds: &[(u64, &[u8])] = &[
+        (0, &[0x00]),
+        (1, &[0x01]),
+        (252, &[0xFC]),
+        (253, &[0xFD, 0xFD, 0x00]),
+        (65535, &[0xFD, 0xFF, 0xFF]),
+    ];
+    for &(seed, encoding) in seeds {
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/rust/crates/rubin-consensus/tests/block_basic_fuzz.rs]: Fixed in 25f9f04: renamed binding to _seed to suppress unused variable warning. CI passes clean with -D warnings.

File: clients/rust/crates/rubin-consensus/src/tests/block_basic.rs [hunk 2]
```diff
@@ -1182,3 +1182,59 @@ fn covenant_genesis_rotation_aware_rejects_non_native_suite() {
     let err = validate_tx_covenants_genesis(&tx, 0, Some(&rp)).unwrap_err();
     assert_eq!(err.code, ErrorCode::TxErrSigAlgInvalid);
 }
+
+// =============================================================
+// RUB-351 hostile tx_count parser hardening — regression tests
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/rust/crates/rubin-consensus/src/tests/block_basic.rs]: Fixed in 25f9f04: section header corrected from RUB-351 to RUB-352 to match the issue tracker reference.

**Comment by 2tbmz9y2xt-lang** [comment]
@codex review
